bitcoin/src/address/error.rs

```diff
@@ -82,20 +80,49 @@ impl std::error::Error for FromScriptError {
     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
         use FromScriptError::*;
 
-        match self {
-            WitnessVersion(e) => Some(e),
-            WitnessProgram(e) => Some(e),
+        match *self {
             UnrecognizedScript => None,
+            WitnessVersion(ref e) => Some(e),
+            WitnessProgram(ref e) => Some(e),
         }
     }
 }
 
 impl From<witness_program::Error> for FromScriptError {
-    fn from(e : witness_program::Error) -> Self { FromScriptError::WitnessProgram(e)}
+    fn from(e : witness_program::Error) -> Self { Self::WitnessProgram(e) }
```

Comment:
This change should not be in this patch but in the one where `FromScriptError` was created. No need to change it now, I just wanted to mention it so you know for next time. 

Something I myself am, at times, at fault of not doing; you want to check your patches before you push and make sure after rebase that all changes are in the correct place and no later patch changes code in a previous patch. That way reviewers can look at the patches in order and not get confused, many of us pull the PR branch down to our local machine and review patch by patch so if things are changed later on in the patch series (the set of commits  making up the PR) it makes review harder.

Thanks for your continued effort, you are doing well, this PR clearly improves the code base.

Reply:
Thank you so much for this. I'll take care of this from next time.